src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -3295,6 +3295,27 @@ void ViewProviderSketch::camSensCB(void* data, SoSensor*)
     vp->onCameraChanged(cam);
 }
 
+Base::Vector3d ViewProviderSketch::getCamCenterInSketchCoordinates(const Gui::View3DInventorViewer* viewer) const
```

Comment:
Actually I have just tried and you can't access it because pImpl is private. So you need to make a function in ViewProviderGridExtension to access it. Basically : 
```
ViewProviderGridExtension:: getCamCenterInSketchCoordinates(){
 return pImpl->getCamCenterInSketchCoordinates();
}
```

Reply:
Or the function can be moved to ViewProviderGridExtension, I don't know if pImpl has access to ViewProviderGridExtension.

Reply:
I wouldn't couple this with the `ViewProviderGridExtension` machinery as the latter, as the name implies, has to do with handling/drawing the grid.
The method name is also misleading as it's not really sketcher-specific.

Perhaps this function could be moved to a common module (where?) and be used by both the axis and the grid sizing mechanisms.

Reply:
Actually ViewProviderSketch is a ViewProviderGridExtension (derives from it). So there no reason you wouldn't access its functions.
Yes the name is not strictly ideal as the gris has been separated from VPSketch. You can rename it something like : getCamCenterIn2DObjectCoordinates.

Reply:
> So there no reason you wouldn't access its functions.

As stated above it's not a technical reason, it just didn't feel right to tangle the grid and the axis behaviour as they are independent entities. Moreover `getCamCenterInSketchCoordinates` depends on internal state (camera center and zoom) that are not updated unless the grid is active so it's not that easy.

Reply:
I see. Well it is indeed harder than it seems.
Then I think the function should be in core in viewInventor

Reply:
Ah but wait I'm dumb, I wrote exactly what you need in my assembly PR : 
View3DInventorViewer::getPointOnXYPlaneOfPlacement
https://github.com/FreeCAD/FreeCAD/pull/10764/files

And this assembly PR is planed to merge very soon (it was to merge yesterday but had a conflict). So if you wait a few days you can then make a new function : 
getCamCenterOnXYPlaneOfPlacement() that pass the cam cetner to getPointOnXYPlaneOfPlacement

Reply:
Awesome news, I'll wait 'till the PR is merged then.

Reply:
@LemonBoy 10764 merged yesterday. If you rebase you'll have access to this function.

Reply:
Thanks for the heads up. I took the chance to address another problem that I've also seen when using the grid, for some "weird" camera orientations the axis length was too short because it didn't take into account the perspective.

The new method seems to work fine, I'm now worried about the computational cost as the camera sensor callback runs continuously.